crates/uv-build/src/lib.rs

```diff
@@ -82,7 +82,9 @@ pub enum Error {
     #[error("Invalid source distribution: {0}")]
     InvalidSourceDist(String),
     #[error("Invalid `pyproject.toml`")]
-    InvalidPyprojectToml(#[from] toml::de::Error),
+    InvalidPyprojectTomlSyntax(#[from] toml_edit::TomlError),
+    #[error("`pyproject.toml` does not match required schema. Note: When using any `[project]` field, at least `project.name` needs to be set.")]
```

Comment:
We only have `name`/`version`/`dynamic` here with the last two being optional, so we skip the check that's it's actually `name` here (in the other case we have to check for name to not confuse it with e.g. invalid `project.dependencies`; it's only a heuristic but imho good enough for the error hinting path)